src/components/Filters/SchemaSieve.ts

```diff
@@ -142,3 +149,125 @@ export const getOperators = (schema: JSONSchema6, field: string) => {
 		label: value.getLabel(subSchema),
 	}));
 };
+
+// Reduces a multi level schema to a single level
+export const flattenSchema = (
+	schema: JSONSchema6,
+	delimiter: string = DELIMITER,
+	parentKey: string = '',
+	accumulator?: JSONSchema6,
```

Comment:
This function works in two ways:

* It might be a pure function that carefully clones & returns data
* It might be a side-effect function that returns data through this argument, and also directly

That's fairly surprising, and makes some of this hard to follow. It'll get way clearer if you split this into two functions:

* One that takes an accumulator, updates it with extra data from a schema, and is potentially recursive
* One that just sets up an accumulator, calls the first, and then returns the accumulator itself

Reply:
Good point, I'll rework this and make it clearer